src/features/orcidlink/ErrorMessage.tsx

```diff
@@ -0,0 +1,36 @@
+import Alert from '@mui/material/Alert';
+import { SerializedError } from '@reduxjs/toolkit';
+import { KBaseBaseQueryError } from '../../common/api/utils/common';
+import styles from './orcidlink.module.scss';
+
+export interface ErrorMessageProps {
+  error: KBaseBaseQueryError | SerializedError;
+}
+
+export default function ErrorMessage({ error }: ErrorMessageProps) {
+  const message = (() => {
+    if ('status' in error) {
+      switch (error.status) {
+        case 'JSONRPC_ERROR':
+          return error.data.error.message;
+        case 'FETCH_ERROR':
+          return 'Fetch Error';
+        case 'CUSTOM_ERROR':
+          return error.error;
+        case 'PARSING_ERROR':
+          return error.error;
+        case 'TIMEOUT_ERROR':
+          return error.error;
+      }
+    } else {
+      return error.message || 'Unknown Error';
+    }
```

Comment:
We tend to avoid `switch` statements. My personal preference is that if we are going to use them then we should make use of their special features, in particular, fall-through and `default` cases. For example:

```js
switch (error.status) {
    case 'JSONRPC_ERROR':
        return error.data.error.message;
    case 'FETCH_ERROR':
        return 'Fetch Error';
    case 'CUSTOM_ERROR':
    case 'PARSING_ERROR':
    case 'TIMEOUT_ERROR':
    default:
        return error.error;
}
```

I think many other `switch` users write code as you have to avoid confusion about these features, and it is this situation that has led us to avoid them on this project. There is another open PR with what I think is [a good alternative](https://github.com/kbase/ui/pull/204/files#diff-10c62a25910e215fb7741ec0dc7cee2e79a1efc7b149536730fbfb748083d842R455-R461).

Reply:
I can abide by the codebase conventions. Yeah, first pass was not to use fallthrough because I hadn't explored those error conditions to see if there was more specialized information before; e.t. timeout in theory should have information about the timeout like the timeout limit and actual elapsed time. But, like I said, I didn't have time on this batch to create a test to exercise the conditions.